### DISCUSSION TITLE
Location in iOS not Working

### DISCUSSION THREAD

**Discussion by zburlott4** [airsdk/Adobe-Runtime-Support#3570] [General]
Attempting to access a users location on iOS doesn't appear to be working. It works on android.  The following is a code sample. Is there something incorrect with this sample?

PList additions

`<key>NSLocationAlwaysUsageDescription</key>
        <string>Your location is required for xyz benefits for you</string>
        <key>NSLocationWhenInUseUsageDescription</key>
        <string>Your location is required for xyz benefits for you</string>
        <key>NSLocationAlwaysAndWhenInUseUsageDescription</key>
        <string>Required by Apple</string>`

And the tracking code

`public function startTracking ():void
      {
         if (Geolocation.permissionStatus.toUpperCase() == "GRANTED"   ) {
            
            if ( Geolocation.isSupported  )
            {
               _geo = new Geolocation();
               _geo.addEventListener(GeolocationEvent.UPDATE, onGeolocationUpdate);
               
               // Optional: Set the desired accuracy (in meters)
               _geo.setRequestedUpdateInterval(3000); // Update every second
            }
            
            
            
         } else {
            trace("Geolocation is not supported on this device.");
         }
         
      } //getCurrentLocation`

**Comment by MalacTheLittle** [comment]
Recently I was doing some project and found out that `Geolocation.permissionStatus` can return `unknown, denied, granted` and `onlyWhenInUse` (last one isn't documented!). If user on iOS tap "Allow Once" or "Allow While Using App" it will be `onlyWhenInUse`. If user in _Settings > Location Services_ selects your app and choose "Always", `permissionStatus` will be `granted`.

Also, you should start with `if(Geolocation.isSupported)` and then check `permissionStatus`. If you get `unknown` or `denied`, you should ask for permission:

```
_geo = new Geolocation();
_geo.addEventListener(PermissionEvent.PERMISSION_STATUS, onPermissionStatus);
_geo.requestPermission();

function onPermissionStatus(event:PermissionEvent):void{
    if(Geolocation.permissionStatus == "granted"){
	
    }else{
	
    }
}
```
One another thing, if user denies Location access to the app two times in a row, `requestPermission` wont trigger Location services dialog anymore.

**Comment by itlancer** [comment]
You can take a look at sample from issue: https://github.com/airsdk/Adobe-Runtime-Support/issues/2943
Also be aware of this issue too: https://github.com/airsdk/Adobe-Runtime-Support/issues/2944

**Reply by zburlott4** [reply]
Request permission is being called, but the permission dialog is not being shown and the app is not showing in the location services of iOS event with the plist permissions. Is there any suggestions on how to diagnose this?
